packages/desktop-client/src/components/transactions/TransactionsTable.tsx

```diff
@@ -949,7 +1008,7 @@ const Transaction = memo(function Transaction({
 
     // Don't save a temporary value (a new payee) which will be
     // filled in with a real id later
-    if (name === 'payee' && value && value.startsWith('new:')) {
+    if (name === 'payee' && value && (value as string).startsWith('new:')) {
```

Comment:
Why is this cast necessary? Naively I would expect the `&& value` to narrow the type to be defined, but maybe I'm missing something.

Reply:
You would think that, wouldn't you? I had the same thought process, and spent about an hour trying to understand Typescript more to figure this out. My understanding is that we are indexing using `TransactionEntity[keyof TransactionEntity]` and the result of that is _all possible types_ which results in some odd behavior. The point being, this was the most straightforward way of handling this and I do know that this is a string... maybe I'll switch to TransactionEntity['payee'] instead.